code/modules/mob/living/carbon/hallucination.dm

```diff
@@ -0,0 +1,1102 @@
+						"[pick("Boarders have", "The AI has", "Intruders have")] demanded we sacrifice a crewmember to them. After [pick("much", "little", "quick")] deliberation, we have chosen [holder]. Please turn yourself over, or [pick("we", "your family", "all of us", "those you love")] will die.",
+						"Central Command has chosen [holder] as the NanoTrasen employee of the month! Everyone please congratulate them.",
+						"Everything is fine.",
+						"The tesla may or may not be loose.",
+						"This is your directive 11. [pick("Spiders have killed several crew.", "Boarders have taken a hostage.", "[holder] is armed and dangerous. Avoid them at all costs.", "Two black-suited individuals have taken items from the vault and armory.")]",
+						"Please stop [pick("drawing in blood. It's unsanitary.", "killing your fellow crew. It's rude.", "[holder] at all costs.", "falling down holes.")]",
+						"[holder] disappoints us all once again.")
+			sound_to(holder, 'sound/misc/announcements/notice.ogg')
+			to_chat(holder, "<h2 class='alert'>Station Announcement</h2>")
+			to_chat(holder, SPAN_ALERT(pick(body)))
+			to_chat(holder, SPAN_ALERT("-[LAZYPICK(hal_sender, holder)]"))
+
+/datum/hallucination/announcement/proc/delam_call()	//for REALLY selling that fake delamination
+	var/list/people = list()
+	for(var/mob/living/carbon/human/M in living_mob_list)
+		if(!M.isMonkey() && !player_is_antag(M, only_offstation_roles = TRUE))		//Antag check prevents meta
+			people += M
+	people -= holder
+	if(!people.len)
+		return
+	var/radio_exclaim = pick("Oh SHIT!", "Oh fuck.", "Uhhh!", "That's not good!", "FUCK.", "Engineering?", "It's under control!", "We're fucked!", "Ohhhh boy.", "What?!", "Um, <b>what?!</b>")
+	to_chat(holder, "<font color='#008000'><b>[pick(people)]</b> says, \"[radio_exclaim]\"</font>")
+
+
+
+/datum/hallucination/pda				//fake PDA messages. this only plays the beep and sends something to chat; it won't show up in the PDA.
+	min_power = 20
+	duration = 900					//this duration length + not allowing duplicates prevents spamming messages on every valid handle_hallucination() which can get VERY annoying if rng decides to give you 3 in a row
+	allow_duplicates = FALSE
+
+/datum/hallucination/pda/start()
+	var/list/sender = message_sender
+	var/mob/living/carbon/human/M = holder
+	for(var/mob/living/carbon/human/H in living_mob_list)
+		if(H.client && !player_is_antag(H, only_offstation_roles = TRUE))	//adds current players to default list to provide variety. leaves out offstation antags.
+			sender += H
+	holder.show_message("<b>Message from [pick(sender)] to [holder.name] ([M.job]),</b> \"[pick(hallucinated_phrases)]\" (<FONT color = blue><u>reply</u></FONT>)")
+	sound_to(holder, 'sound/machines/twobeep.ogg')
+
+
+/datum/hallucination/paranoia
+
+/datum/hallucination/paranoia/can_affect(mob/living/carbon/C)
+	if(!..())
+		return FALSE
+	for(var/mob/living/M in oview(C))
+		return TRUE
```

Comment:
What's this for?

Reply:
Paranoia has you hallucinate a mob in view doing some emote. If there's no mob in view, it should fail the can_affect() check so it doesn't get added to the list of candidates. If it was picked it would do nothing.